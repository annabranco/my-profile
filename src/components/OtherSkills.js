import React from 'react';
import '../styles/components/OtherSkills.css';
import MacBar from '../images/bg/bg-mac.jpg';

class OtherSkills extends React.Component {

	constructor(props) {
		super(props)

		this.state = {
			opened: false,
			read: false
		}
	}


		open = () => {
			if (!this.state.opened) {

				this.setState({
					opened: true,
					read: true
				 });
				document.querySelector('.section__otherSkills').classList.add('opened');
				document.querySelector('.otherSkills__mac').classList.add('opened');
				document.querySelector('.otherSkills__mac--screen').classList.add('opened');
				document.querySelector('.otherSkills__mac--keyboard').classList.add('hidden');
				document.querySelector('.otherSkills__mac--keyboard-keyArea').classList.add('hidden');
				this.props.markAsRead();

			} else {

				this.setState({ opened: false });
				document.querySelector('.section__otherSkills').classList.remove('opened');
				document.querySelector('.otherSkills__mac').classList.remove('opened');
				document.querySelector('.otherSkills__mac--screen').classList.remove('opened');
				document.querySelector('.otherSkills__mac--keyboard').classList.remove('hidden');
				document.querySelector('.otherSkills__mac--keyboard-keyArea').classList.remove('hidden');
			}
		}

	render () {

		const text = this.props.texts.OtherSkills;
		const language = this.props.language;

		return (

			<React.Fragment>

				{!this.props.viewedOtherSkills && !this.state.read ?
					<React.Fragment>
						<p className="seabed__findSomething">{text[language].find}</p>
						<p className="seabed__findSomething">{text[language].find2}</p>
						<p className="seabed__findSomething">{text[language].investigate}</p>
					</React.Fragment>
				: null
				}

				<div className="otherSkills__mac">
					<div className="otherSkills__mac--screen">

						<section className="section__otherSkills" onClick={this.open}>

							<img src={MacBar} alt="" className="skills--topBar"/>
							<p className="seabed__click2close seabed__click2close-otherSkills">{this.props.texts.Seabed[language].click2close}</p>

							<div className="skills__outer skills__outer--languages">
								<h2 className="skills__table--title">{text[language].languages}</h2>
								<table className="skills__table skills__table--languages">
									<tbody>
										<tr>
										    <td>{text[language].english}<img src="https://www.countryflags.io/us/flat/16.png" alt="" className="languages--flag" /></td>
										    <td><i className="fas fa-comments icon--languages" title="Speaking"></i>{text[language].fluent}</td>
										    <td><i className="fas fa-book-open icon--languages" title="Reading"></i>{text[language].fluent}</td>
											<td><i className="fas fa-pen icon--languages" title="Writing"></i>{text[language].fluent}</td>
										</tr>
										<tr>
										    <td>{text[language].spanish}<img src="https://www.countryflags.io/es/flat/16.png" alt="" className="languages--flag" /></td>
										    <td><i className="fas fa-comments icon--languages" title="Speaking"></i>{text[language].fluent}</td>
										    <td><i className="fas fa-book-open icon--languages" title="Reading"></i>{text[language].fluent}</td>
											<td><i className="fas fa-pen icon--languages" title="Writing"></i>{text[language].fluent}</td>
										</tr>
										<tr>
											<td>{text[language].portuguese}<img src="https://www.countryflags.io/br/flat/16.png" alt="" className="languages--flag" /></td>
											<td><i className="fas fa-comments icon--languages" title="Speaking"></i>{text[language].fluent}</td>
											<td><i className="fas fa-book-open icon--languages" title="Reading"></i>{text[language].fluent}</td>
											<td><i className="fas fa-pen icon--languages" title="Writing"></i>{text[language].fluent}</td>
										</tr>
										<tr>
											<td>{text[language].french}<img src="https://www.countryflags.io/fr/flat/16.png" alt="" className="languages--flag" /></td>
											<td><i className="fas fa-comments icon--languages" title="Speaking"></i>{text[language].intermediate}</td>
											<td><i className="fas fa-book-open icon--languages" title="Reading"></i>{text[language].intermediate}</td>
											<td><i className="fas fa-pen icon--languages" title="Writing"></i>{text[language].basic}</td>
										</tr>
									 	<tr>
											<td>{text[language].russian}<img src="https://www.countryflags.io/ru/flat/16.png" alt="" className="languages--flag" /></td>
											<td><i className="fas fa-comments icon--languages" title="Speaking"></i>{text[language].basic}</td>
											<td><i className="fas fa-book-open icon--languages" title="Reading"></i>{text[language].basic}</td>
											<td><i className="fas fa-pen icon--languages" title="Writing"></i>{text[language].basic}</td>
									 	</tr>
									</tbody>
								</table>
							</div>

							<div className="skills__outer skills__outer--other">
								<h2 className="skills__table--title">{text[language].other}</h2>
								<table className="skills__table skills__table--other">
									<thead>
										<tr>
											<th>{text[language].skill}</th>
											<th>{text[language].how}</th>
										</tr>
									</thead>
									<tbody>
										<tr>
											<td>{text[language].skill1}</td>
											<td>{text[language].skill1details}</td>
										</tr>
										<tr>
											<td>{text[language].skill2}</td>
											<td>{text[language].skill2details}</td>
										</tr>
										<tr>
											<td>{text[language].skill3}</td>
											<td>{text[language].skill3details}</td>
										</tr>
										<tr>
											<td>{text[language].skill4}</td>
											<td>{text[language].skill4details}</td>
										</tr>
										<tr>
											<td>{text[language].skill5}</td>
											<td>{text[language].skill5details}</td>
										</tr>
									</tbody>
								</table>
							</div>
							<div className="skills__outer skills__outer--design">
								<h2 className="skills__table--title">{text[language].design}</h2>
								<p className="skills__design">{text[language].samples}</p>
								<div className="skills__design--samples">
								<img src="https://via.placeholder.com/300x200/fff5f4/000?text=sample" alt="" className="design__sample"/>
								<img src="https://via.placeholder.com/300x200/fff5f4/000?text=sample" alt="" className="design__sample"/>
								<img src="https://via.placeholder.com/300x200/fff5f4/000?text=sample" alt="" className="design__sample"/>
								<img src="https://via.placeholder.com/300x200/fff5f4/000?text=sample" alt="" className="design__sample"/>
								<img src="https://via.placeholder.com/300x200/fff5f4/000?text=sample" alt="" className="design__sample"/>
								<img src="https://via.placeholder.com/300x200/fff5f4/000?text=sample" alt="" className="design__sample"/>
								</div>
							</div>


						</section>
					</div>
					<div className="otherSkills__mac--keyboard"></div>
					<div className="otherSkills__mac--keyboard-keyArea">
						<p className="otherSkills__mac--keyboard-keys"> . . . . . . . ...</p>
						<p className="otherSkills__mac--keyboard-keys">... . . . . . . ..</p>
						<p className="otherSkills__mac--keyboard-keys">. . . . . . . . . .</p>
						<p className="otherSkills__mac--keyboard-keys">. . . . . . . . . .</p>
						<p className="otherSkills__mac--keyboard-keys">--------- ......</p>
					</div>
				</div>

			</React.Fragment>

		);
	}
}

export default OtherSkills;
