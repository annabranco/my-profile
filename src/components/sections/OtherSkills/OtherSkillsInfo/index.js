import React from 'react';
import { useSelector } from 'react-redux';
import { bool, func } from 'prop-types';
import {
  globalTextsSelector,
  otherSkillsTextSelector
} from '../../../../redux/selectors';
import {
  CloseButton,
  DesignArea,
  DesignText,
  Flag,
  Icon,
  LanguagesArea,
  LanguagesTable,
  OtherInformationAreaArea,
  OtherInformationAreaTable,
  OtherSkillsWrapper,
  Sample,
  SamplesWrapper,
  Title
} from './styles';
import { getFlagURL } from '../../../../utils/icons';
import { isDesktop } from '../../../../utils/device';

const OtherSkillsInfo = ({ onClickOpen, onClickClose, visible }) => {
  const texts = useSelector(otherSkillsTextSelector);
  const globalTexts = useSelector(globalTextsSelector);

  return (
    <OtherSkillsWrapper
      aria-label={globalTexts.open}
      onClick={() => onClickOpen('otherSkillsSection')}
      role="button"
      tabIndex={0}
      visible={visible}
    >
      <LanguagesArea>
        <Title>{texts.languages}</Title>
        <LanguagesTable>
          <tbody>
            {texts.languagesSpoken.map(item => (
              <tr key={item.language}>
                <td>
                  {item.language}
                  <Flag
                    src={getFlagURL({
                      country: item.flagCode,
                      style: 'round3d',
                      size: 'small'
                    })}
                    alt=""
                  />
                </td>
                <td>
                  <Icon className="fas fa-comments" title="Speaking" />
                  {item.speak}
                </td>
                <td>
                  <Icon className="fas fa-book-open" title="Reading" />
                  {item.read}
                </td>
                <td>
                  <Icon className="fas fa-pen" title="Writing" />
                  {item.write}
                </td>
              </tr>
            ))}
          </tbody>
        </LanguagesTable>
      </LanguagesArea>

      <OtherInformationAreaArea>
        <Title>{texts.other}</Title>
        <OtherInformationAreaTable>
          {isDesktop && (
            <thead>
              <tr>
                <th>{texts.skill}</th>
                <th>{texts.how}</th>
              </tr>
            </thead>
          )}
          <tbody>
            {texts.skills.map(item => (
              <tr key={item.skill}>
                <td>{item.skill}</td>
                <td>{item.details}</td>
              </tr>
            ))}
          </tbody>
        </OtherInformationAreaTable>
      </OtherInformationAreaArea>
      {isDesktop && (
        <>
          <DesignArea>
            <Title>{texts.design}</Title>
            <DesignText>{texts.samples}</DesignText>
            <SamplesWrapper>
              <a
                href="https://pre00.deviantart.net/f5fe/th/pre/f/2018/065/a/2/rio_de_janeiro__um_retrato_da_violencia_by_annabranco-dc53j1i.jpg"
                rel="noopener noreferrer"
                target="_Blank"
              >
                <Sample image="https://pre00.deviantart.net/f5fe/th/pre/f/2018/065/a/2/rio_de_janeiro__um_retrato_da_violencia_by_annabranco-dc53j1i.jpg" />
              </a>
              <a
                href="https://img00.deviantart.net/9fb7/i/2018/069/2/c/recuerda_siempre_mirar_el_color_de_las_banderas_by_annabranco-dc5gv8e.jpg"
                rel="noopener noreferrer"
                target="_Blank"
              >
                <Sample image="https://img00.deviantart.net/9fb7/i/2018/069/2/c/recuerda_siempre_mirar_el_color_de_las_banderas_by_annabranco-dc5gv8e.jpg" />
              </a>
              <a
                href="https://pre00.deviantart.net/3afa/th/pre/i/2018/069/e/3/basic_english_by_annabranco-dc5go6q.jpg"
                rel="noopener noreferrer"
                target="_Blank"
              >
                <Sample image="https://pre00.deviantart.net/3afa/th/pre/i/2018/069/e/3/basic_english_by_annabranco-dc5go6q.jpg" />
              </a>
              <a
                href="https://pre00.deviantart.net/6e8f/th/pre/f/2018/245/e/8/violencia1_en_by_annabranco-dclta8q.jpg"
                rel="noopener noreferrer"
                target="_Blank"
              >
                <Sample image="https://pre00.deviantart.net/6e8f/th/pre/f/2018/245/e/8/violencia1_en_by_annabranco-dclta8q.jpg" />
              </a>
              <a
                href="https://pre00.deviantart.net/9f40/th/pre/f/2018/065/1/7/terremoto_en_mexico_19_09_17_by_alvarobranco_dbo6p_by_annabranco-dc53k8f.jpg"
                rel="noopener noreferrer"
                target="_Blank"
              >
                <Sample image="https://pre00.deviantart.net/9f40/th/pre/f/2018/065/1/7/terremoto_en_mexico_19_09_17_by_alvarobranco_dbo6p_by_annabranco-dc53k8f.jpg" />
              </a>
              <a
                href="https://pre00.deviantart.net/d51e/th/pre/i/2018/069/5/b/european_map__arabic__by_annabranco-dc5gve5.jpg"
                rel="noopener noreferrer"
                target="_Blank"
              >
                <Sample image="https://pre00.deviantart.net/d51e/th/pre/i/2018/069/5/b/european_map__arabic__by_annabranco-dc5gve5.jpg" />
              </a>
            </SamplesWrapper>
          </DesignArea>
          <CloseButton
            aria-label={globalTexts.close}
            hidden={!visible}
            onClick={() => onClickClose('otherSkillsSection')}
            type="button"
          >
            {globalTexts.close}
          </CloseButton>
        </>
      )}
    </OtherSkillsWrapper>
  );
};

OtherSkillsInfo.propTypes = {
  onClickOpen: func,
  onClickClose: func,
  visible: bool.isRequired
};

OtherSkillsInfo.defaultProps = {
  onClickOpen: () => null,
  onClickClose: () => null
};

export default OtherSkillsInfo;
