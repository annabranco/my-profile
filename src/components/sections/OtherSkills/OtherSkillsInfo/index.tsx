import { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { bool, func } from 'prop-types';
import { IGlobalTexts, IOtherSkillsText } from '../../../../types/interfaces';
import {
  globalTextsSelector,
  otherSkillsTextSelector
} from '../../../../redux/selectors';
import { getFlagURL } from '../../../../utils/icons';
import { isDesktop } from '../../../../utils/device';
import {
  DesignArea,
  DesignText,
  Flag,
  Icon,
  LanguagesArea,
  LanguagesTable,
  OtherInformationAreaArea,
  OtherInformationAreaTable,
  OtherSkillsWrapper,
  Sample,
  SamplesWrapper,
  Title
} from './styles';

interface Props {
  onClickClose: (type: string) => void;
  onClickOpen: (type: string) => void;
  visible: boolean;
}

const OtherSkillsInfo = ({
  onClickOpen,
  onClickClose,
  visible
}: Props): ReactElement => {
  const texts: IOtherSkillsText = useSelector(otherSkillsTextSelector);
  const globalTexts: IGlobalTexts = useSelector(globalTextsSelector);

  return (
    <OtherSkillsWrapper
      aria-label={globalTexts.open}
      onClick={() =>
        visible
          ? onClickClose('otherSkillsSection')
          : onClickOpen('otherSkillsSection')
      }
      role="button"
      tabIndex={0}
      visible={visible}
    >
      {visible && (
        <>
          <LanguagesArea>
            <Title>{texts.languages}</Title>
            <LanguagesTable>
              <tbody>
                {texts.languagesSpoken.map(item => (
                  <tr key={item.language}>
                    <td>
                      {item.language}
                      <Flag
                        src={getFlagURL({
                          country: item.flagCode,
                          size: 'small',
                          style: 'round3d'
                        })}
                        alt=""
                      />
                    </td>
                    <td>
                      <Icon className="fas fa-comments" title="Speaking" />
                      {item.speak}
                    </td>
                    <td>
                      <Icon className="fas fa-book-open" title="Reading" />
                      {item.read}
                    </td>
                    <td>
                      <Icon className="fas fa-pen" title="Writing" />
                      {item.write}
                    </td>
                  </tr>
                ))}
              </tbody>
            </LanguagesTable>
          </LanguagesArea>

          <OtherInformationAreaArea>
            <Title>{texts.other}</Title>
            <OtherInformationAreaTable>
              <tbody>
                {texts.skills.map(item => (
                  <tr key={item.skill}>
                    <td>{item.skill}</td>
                    <td>{item.details}</td>
                  </tr>
                ))}
              </tbody>
            </OtherInformationAreaTable>
          </OtherInformationAreaArea>
          {isDesktop && (
            <>
              <DesignArea onClick={event => event.stopPropagation()}>
                <Title>{texts.design}</Title>
                <DesignText>{texts.samples}</DesignText>
                <SamplesWrapper>
                  <a
                    href="https://pre00.deviantart.net/f5fe/th/pre/f/2018/065/a/2/rio_de_janeiro__um_retrato_da_violencia_by_annabranco-dc53j1i.jpg"
                    rel="noopener noreferrer"
                    target="_Blank"
                  >
                    <Sample image="https://pre00.deviantart.net/f5fe/th/pre/f/2018/065/a/2/rio_de_janeiro__um_retrato_da_violencia_by_annabranco-dc53j1i.jpg" />
                  </a>
                  <a
                    href="https://img00.deviantart.net/9fb7/i/2018/069/2/c/recuerda_siempre_mirar_el_color_de_las_banderas_by_annabranco-dc5gv8e.jpg"
                    rel="noopener noreferrer"
                    target="_Blank"
                  >
                    <Sample image="https://img00.deviantart.net/9fb7/i/2018/069/2/c/recuerda_siempre_mirar_el_color_de_las_banderas_by_annabranco-dc5gv8e.jpg" />
                  </a>
                  <a
                    href="https://pre00.deviantart.net/3afa/th/pre/i/2018/069/e/3/basic_english_by_annabranco-dc5go6q.jpg"
                    rel="noopener noreferrer"
                    target="_Blank"
                  >
                    <Sample image="https://pre00.deviantart.net/3afa/th/pre/i/2018/069/e/3/basic_english_by_annabranco-dc5go6q.jpg" />
                  </a>
                  <a
                    href="https://pre00.deviantart.net/6e8f/th/pre/f/2018/245/e/8/violencia1_en_by_annabranco-dclta8q.jpg"
                    rel="noopener noreferrer"
                    target="_Blank"
                  >
                    <Sample image="https://pre00.deviantart.net/6e8f/th/pre/f/2018/245/e/8/violencia1_en_by_annabranco-dclta8q.jpg" />
                  </a>
                  <a
                    href="https://pre00.deviantart.net/9f40/th/pre/f/2018/065/1/7/terremoto_en_mexico_19_09_17_by_alvarobranco_dbo6p_by_annabranco-dc53k8f.jpg"
                    rel="noopener noreferrer"
                    target="_Blank"
                  >
                    <Sample image="https://pre00.deviantart.net/9f40/th/pre/f/2018/065/1/7/terremoto_en_mexico_19_09_17_by_alvarobranco_dbo6p_by_annabranco-dc53k8f.jpg" />
                  </a>
                  <a
                    href="https://pre00.deviantart.net/d51e/th/pre/i/2018/069/5/b/european_map__arabic__by_annabranco-dc5gve5.jpg"
                    rel="noopener noreferrer"
                    target="_Blank"
                  >
                    <Sample image="https://pre00.deviantart.net/d51e/th/pre/i/2018/069/5/b/european_map__arabic__by_annabranco-dc5gve5.jpg" />
                  </a>
                </SamplesWrapper>
              </DesignArea>
            </>
          )}
        </>
      )}
    </OtherSkillsWrapper>
  );
};

OtherSkillsInfo.propTypes = {
  onClickOpen: func,
  onClickClose: func,
  visible: bool.isRequired
};

OtherSkillsInfo.defaultProps = {
  onClickOpen: () => null,
  onClickClose: () => null
};

export default OtherSkillsInfo;
