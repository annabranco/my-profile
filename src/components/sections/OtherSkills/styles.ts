import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlack,
  colorGrayNormal,
  colorGrayDark,
  colorRedBright,
  fontSubtitle
} from '../../../styles/theme';
import { Hidden, NotDisplayed } from '../../../styles/global';

export const KeyboardKeysWrapper = styled.div`
  position: absolute;
  top: 58px;
  left: 25px;
  opacity: 0.5;
  width: 70px;

  ${({ hidden }) =>
    hidden &&
    css`
      ${NotDisplayed}
    `}
`;
KeyboardKeysWrapper.displayName = 'Keyboard Keys Wrapper';

export const Keys = styled.div`
  transform: perspective(0em) rotateX(120deg) skewX(100deg);
  margin: -10px 0;
  letter-spacing: -1.3px;
  line-height: 0.9;
  text-align: center;
`;
Keys.displayName = 'Keyboard Keys';

export const MacCamera = styled.div`
  z-index: 100;
  position: absolute;
  top: -25px;
  left: 50%;
  transform: translate(-50%, 0);
  width: 20px;
  height: 20px;
  background: #2f353f;
  border-radius: 50px;
  border: 2px solid ${rgba('#2f353f', 0.1)};
  box-shadow: inset 0 0 1px ${rgba('black', 0.1)};

  &::before {
    content: '';
    position: absolute;
    top: 2px;
    left: 50%;
    width: 5px;
    height: 5px;
    border-radius: 50px;
    background-color: white;
    opacity: 0.3;
    box-shadow: 0 0 5px 3px white;
  }

  @media all and (min-width: 2000px) {
    top: -35px;
  }
`;
MacCamera.displayName = 'Keyboard MacCamera';

interface MacCloseButtonProps extends React.ComponentPropsWithoutRef<'button'> {
  isTabButton?: boolean;
  visible?: boolean;
}

export const MacCloseButton = styled.button<MacCloseButtonProps>`
  z-index: 10;
  position: fixed;
  top: 43px;
  left: 6px;
  background: ${colorRedBright};
  border: 0;
  border-radius: 50%;
  height: 13px;
  width: 13px;
  font-size: 0.6rem;
  color: ${colorBlack};
  cursor: pointer;
  outline: none !important;
  ${Hidden}

  ${({ visible }) =>
    visible &&
    css`
      visibility: visible;
    `}

  &::after {
    content: 'x';
    transform: translate(-3px, -3px);
    display: block;
    font-size: 1rem;
    font-family: monospace;
  }

  &:hover {
    color: ${colorWhite};
  }

  ${({ isTabButton }) =>
    isTabButton &&
    css`
      top: 47px;
      left: 570px;
      background: #f2f2f2;
      font-weight: bolder;

      &::after {
        content: 'x';
        transform: translate(-3px, -3px);
        display: block;
        font-size: 1.2rem;
        font-family: monospace;
      }
      &:hover {
        color: ${colorRedBright};
      }
    `}
`;

interface MacKeyboardProps extends React.ComponentPropsWithoutRef<'div'> {
  hidden?: boolean;
}

export const MacKeyboard = styled.div<MacKeyboardProps>`
  position: absolute;
  top: 30px;
  left: 2px;
  opacity: 0.6;
  transform: perspective(9em) rotateX(65deg) skewX(15deg);
  border-bottom: 3px solid ${colorBlack};
  border-left: 1px solid ${colorBlack};
  background: ${colorGrayNormal};

  height: 55px;
  width: 77px;
  box-shadow: -2px 2px 10px 3px ${rgba(colorGrayDark, 0.8)};

  ${({ hidden }) =>
    hidden &&
    css`
      ${NotDisplayed}
    `}
`;
MacKeyboard.displayName = 'Mac Keyboard';

interface MacScreenProps extends React.ComponentPropsWithoutRef<'div'> {
  visible?: boolean;
}

export const MacScreen = styled.div<MacScreenProps>`
  opacity: 0.9;
  transform: perspective(8em) rotateX(-5deg) skewX(-4deg);
  border-left: 1px solid ${colorGrayNormal};
  background: ${colorBlack};
  height: 50px;
  width: 70px;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 1;
      transform: none;
      border-left: 0;
      background: ${colorWhite};
      height: ${`calc(${window.innerHeight}px - 60px)`};
      width: 100%;
      overflow: hidden;
    `}

  @media all and (min-width: 2000px) {
    ${({ visible }) =>
      visible &&
      css`
        height: ${`calc(${window.innerHeight}px - 120px)`};
      `}
  }
`;
MacScreen.displayName = 'Mac Screen';

export const Mark = styled.div`
  z-index: 100;
  position: absolute;
  bottom: -25px;
  left: 50%;
  transform: translate(-50%, 0);
  color: white;
  font-weight: 500;
  font-family: ${fontSubtitle};
  font-size: 0.9rem;
  opacity: 0.8;
  letter-spacing: 1px;
  user-select: none;

  @media all and (min-width: 2000px) {
    bottom: -40px;
  }
`;
Mark.displayName = 'Keyboard Mark';

export const MarkLight = styled.span`
  font-weight: 100;
`;
MarkLight.displayName = 'Keyboard MarkLight';

interface SectionOtherSkillsProps
  extends React.ComponentPropsWithoutRef<'section'> {
  visible?: boolean;
}

export const SectionOtherSkills = styled.section<SectionOtherSkillsProps>`
  z-index: 6;
  position: absolute;
  top: 80vh;
  left: 60%;
  opacity: 0.8;
  height: 50px;
  width: 70px;

  @media all and (min-width: 768px) {
    left: 40%;

    ${({ visible }) =>
      visible &&
      css`
        top: 0;
        left: 0;
        opacity: 1;
        height: auto;
        width: 101%;
        border: 30px solid #1e2126;
      `}
  }
  @media all and (min-width: 1400px) {
    ${({ visible }) =>
      visible &&
      css`
        width: 100%;
      `}
  }

  @media all and (min-width: 2000px) {
    ${({ visible }) =>
      visible &&
      css`
        border: 60px solid #1e2126;
      `}
  }
`;
SectionOtherSkills.displayName = 'OTHER SKILLS Section';

interface TopBarProps extends React.ComponentPropsWithoutRef<'img'> {
  visible?: boolean;
}

export const TopBar = styled.img<TopBarProps>`
  z-index: 9;
  position: absolute;
  top: 0;
  width: 60px;
  opacity: 0;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 1;
      z-index: 4;
      position: fixed;
      top: 30px;
      left: 30px;
      width: 96.3vw;
    `}

  @media all and (min-width: 2000px) {
    top: 60px;
    left: 60px;
    width: 95.5vw;
  }
`;
TopBar.displayName = 'Top Bar';
