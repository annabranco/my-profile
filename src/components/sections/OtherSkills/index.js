import React from 'react';
import { useSelector } from 'react-redux';
import { func } from 'prop-types';
import { otherSkillsTextSelector } from '../../../redux/selectors';
import { isDesktop } from '../../../utils/device';
import OtherSkillsInfo from './OtherSkillsInfo';
import { MacNotebook } from '../../../assets/images';
import { seabedElementsPropType } from '../../../types/propTypes';
import { TextFindSomething } from '../Seabed/styles';
import {
  KeyboardKeysWrapper,
  Keys,
  MacKeyboard,
  MacScreen,
  SectionOtherSkills,
  TopBar
} from './styles';

const OtherSkills = ({
  onClickClose,
  onClickOpen,
  status: { read, visible }
}) => {
  const texts = useSelector(otherSkillsTextSelector);

  return (
    <>
      {!read && isDesktop ? (
        <>
          <TextFindSomething>{texts.find}</TextFindSomething>
          <TextFindSomething>{texts.find2}</TextFindSomething>
          <TextFindSomething>{texts.investigate}</TextFindSomething>
        </>
      ) : null}

      <SectionOtherSkills id="Other Skills Section" visible={visible}>
        {isDesktop && <TopBar visible={visible} src={MacNotebook} alt="" />}
        <MacScreen visible={visible}>
          <OtherSkillsInfo
            onClickOpen={onClickOpen}
            onClickClose={onClickClose}
            visible={visible}
          />
        </MacScreen>
        <MacKeyboard hidden={visible} />
        <KeyboardKeysWrapper hidden={visible}>
          <Keys> . . . . . . . ...</Keys>
          <Keys>... . . . . . . ..</Keys>
          <Keys>. . . . . . . . . .</Keys>
          <Keys>. . . . . . . . . .</Keys>
          <Keys>------......</Keys>
        </KeyboardKeysWrapper>
      </SectionOtherSkills>
    </>
  );
};

OtherSkills.propTypes = {
  onClickClose: func.isRequired,
  onClickOpen: func.isRequired,
  status: seabedElementsPropType.isRequired
};

export default OtherSkills;
