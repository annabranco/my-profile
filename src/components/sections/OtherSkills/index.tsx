import React, { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { func } from 'prop-types';
import { ISeabedElements } from '../../../types/interfaces';
import { otherSkillsTextSelector } from '../../../redux/selectors';
import { isDesktop } from '../../../utils/device';
import OtherSkillsInfo from './OtherSkillsInfo';
import { MacNotebook } from '../../../assets/images';
import { seabedElementsPropType } from '../../../types/propTypes';
import { TextFindSomething } from '../Seabed/styles';
import {
  KeyboardKeysWrapper,
  Keys,
  MacKeyboard,
  MacScreen,
  SectionOtherSkills,
  TopBar
} from './styles';

interface Props {
  onClickClose: (type: string) => void;
  onClickOpen: (type: string) => void;
  status: ISeabedElements;
}

const OtherSkills = ({
  onClickClose,
  onClickOpen,
  status: { read, visible }
}: Props): ReactElement => {
  const texts = useSelector(otherSkillsTextSelector);

  return (
    <>
      {!read && isDesktop ? (
        <>
          <TextFindSomething>{texts.find}</TextFindSomething>
          <TextFindSomething>{texts.find2}</TextFindSomething>
          <TextFindSomething>{texts.investigate}</TextFindSomething>
        </>
      ) : null}

      <SectionOtherSkills id="Other Skills Section" visible={visible}>
        {isDesktop && <TopBar visible={visible} src={MacNotebook} alt="" />}
        <MacScreen visible={visible}>
          <OtherSkillsInfo
            onClickOpen={onClickOpen}
            onClickClose={onClickClose}
            visible={visible}
          />
        </MacScreen>
        <MacKeyboard hidden={visible} />
        <KeyboardKeysWrapper hidden={visible}>
          <Keys> . . . . . . . ...</Keys>
          <Keys>... . . . . . . ..</Keys>
          <Keys>. . . . . . . . . .</Keys>
          <Keys>. . . . . . . . . .</Keys>
          <Keys>------......</Keys>
        </KeyboardKeysWrapper>
      </SectionOtherSkills>
    </>
  );
};

OtherSkills.propTypes = {
  onClickClose: func.isRequired,
  onClickOpen: func.isRequired,
  status: seabedElementsPropType.isRequired
};

export default OtherSkills;
