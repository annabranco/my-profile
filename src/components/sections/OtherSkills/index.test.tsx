import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import '../../../tests/mocks/matchMedia';
import OtherSkills from '.';
import { setupMockProvider, subSectionsStatus } from '../../../tests/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< OtherSkills >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <OtherSkills
          onClickClose={() => null}
          onClickOpen={() => null}
          status={subSectionsStatus}
        />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should render an event message on desktops', () => {});

  xit('should display a section title', () => {});

  xit('should display a notebook element', () => {});

  xit('should call prop onClickOpen when notebook is clicked', () => {});

  xit('should display a SectionOtherSkills component passing the visibility prop', () => {});

  xit('should render a Mac interface simulation on desktops when prop visible is true', () => {});

  xit('should translate all texts when the language is changed', () => {});
});
