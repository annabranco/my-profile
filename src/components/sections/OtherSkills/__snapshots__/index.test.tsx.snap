// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< OtherSkills > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OtherSkills
    onClickClose={[Function]}
    onClickOpen={[Function]}
    status={
      Object {
        "active": false,
        "read": false,
        "visible": false,
      }
    }
  />
</Provider>
`;
